src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -189,5 +180,91 @@ public static string FormatLiteral(char c, bool quote)
                 (quote ? ObjectDisplayOptions.UseQuotes : ObjectDisplayOptions.None);
             return ObjectDisplay.FormatLiteral(c, options);
         }
+
+        /// <summary>
+        /// Returns a textual representation of an object of primitive type as an array of string parts,
+        /// each of which has a kind. Useful for colorizing the display string.
+        /// </summary>
+        /// <param name="obj">A value to display as string parts.</param>
+        /// <param name="options">Specifies the display options.</param>
+        /// <returns>A list of display parts (or <see langword="default"/> if the type is not supported).</returns>
+        /// <remarks>
+        /// Handles <see cref="bool"/>, <see cref="string"/>, <see cref="char"/>, <see cref="sbyte"/>
+        /// <see cref="byte"/>, <see cref="short"/>, <see cref="ushort"/>, <see cref="int"/>, <see cref="uint"/>,
+        /// <see cref="long"/>, <see cref="ulong"/>, <see cref="double"/>, <see cref="float"/>, <see cref="decimal"/>,
+        /// and <see langword="null"/>.
+        /// </remarks>
```

Comment:
1. just to check, but it's intentional that dates are not in this list?
2. this list doesn't mention enums, but i can see that is handled in teh check below